Rituals and Psycasts update

MaxDorob opened the pull request.

MaxDorob committed:
Spawn bloodstones on dead prisoner

MaxDorob committed:
Apply -> ApplyExtraOutcome

MaxDorob committed:
Merge branch 'SacrificeScore' into RitualsRework

MaxDorob committed:
Score-based result for result

MaxDorob committed:
We don't need the ideology-moralist at our party, right?

MaxDorob committed:
Revia count affects ritual quality

MaxDorob committed:
Do not spawn reward if sacrificed pawn got resurected (e.g. Archon)

MaxDorob committed:
Reduced last stage duration to few ticks (not sure why, but I can't remove the stage without failing ritual)

MaxDorob committed:
Revia counter improvment

MaxDorob committed:
Somehow working soulreap blessing ritual

MaxDorob committed:
Show required count in description

MaxDorob committed:
Pawn able to use interaction position on building

MaxDorob committed:
typo

MaxDorob committed:
Sacrificer will stay right on prisoner (spams errors tho)

MaxDorob committed:
Revert "Sacrificer will stay right on prisoner (spams errors tho)"

This reverts commit 8412e46c0ac129190b54a7a23b6ef7a077a0fb20.

MaxDorob committed:
Sacrificer stays on prisoner

MaxDorob committed:
Debug things

MaxDorob committed:
Sacrifice positions tweaks

MaxDorob committed:
Fixed take bloodstones job

MaxDorob committed:
Cleanup

Commit:
forgot to change roleId

Commit:
rename variable

Commit:
cleanup

Commit:
Missing nameMaker fix

Commit:
Temp non-Ideology icon

Commit:
Xenotype conversation ritual (not completed yet)

Commit:
Corrected convertee pos

Commit:
splited jobGiver

Commit:
sacrificer takes 10 bloodstones before xenotype change ritual

Commit:
Corrected RitualRole folder

Commit:
Sacrificer uses bloodstones for convertion

Commit:
Prerequest for conversation ritual

Commit:
xenogerm coma as outcome

Commit:
Psycast ritual

Commit:
Duration

Commit:
various changes

Commit:
don't use Repeat Penalty

Commit:
Precepts cleanup

Commit:
cleanup

Commit:
Outcomes cleanuo

Commit:
Changed kill info

Commit:
classic precept

Commit:
Revia count impact capped at 9

Commit:
Quality changes with tails count (soulreap only now)

Commit:
Skarne's sculpture for rituals

Commit:
Fail trigger

Commit:
Dll for tests

Commit:
Ideology precepts

Commit:
use sacrifice precepts instead of execution

Commit:
Text (thanks Dr. Nolegs) and various balance changes

Commit:
1.6 merge

Commit:
Precepts moved in 1.6

Commit:
desc changes

Commit:
Moved rituals to 1.6

Commit:
Moved duties to 1.6

Commit:
Thoughts on bless

Commit:
Removed Ideology from 1.5 and moved to 1.6

Commit:
Fixed precept name

Commit:
Art is 1.6 only

Commit:
1.6 only defs

Commit:
Reverted 1.5 meme changes. Moved to 1.6

Commit:
1.6 only

Commit:
Allow some rituals on spot

Commit:
Don't change color of Revia rituals

Commit:
Arguments

Commit:
ref version

Commit:
Soulreap blessing ritual icon

Commit:
Corrected MayRequire

Commit:
More fixes

Commit:
Psycast blessing (requires Royalty)

MaxDorob committed:
Reduced duration

MaxDorob committed:
Skarne's sculptrure

MaxDorob committed:
Missing def fix

Commit:
Merge branch 'RitualsRework' of https://github.com/MaxDorob/RimWorld-ReviaRaceMod into RitualsRework

Commit:
Added slider to dialog

MaxDorob committed:
Slider to top

MaxDorob committed:
Various slider changes

MaxDorob committed:
Working version

Commit:
Job driver various changes

MaxDorob committed:
Use xenotype icon as ritual icon

MaxDorob committed:
Removed eyes from xenotype icon

MaxDorob committed:
Serrated dagger as ritual icon

Commit:
Corrected job giver

Commit:
driver fix

Commit:
Create curve from settings

Commit:
Merge branch 'RitualsRework_BloodstonesCountSlider' into RitualsRework

Commit:
Removed seat effect

Commit:
NRE fix

Commit:
Reduced fail chance

Commit:
debug changes

Commit:
art changes

Commit:
Changed tails factor

Commit:
Curve calc changed

Commit:
Increased chance

Commit:
Another rebalance

Commit:
Post load NRE fix

Commit:
Minimal required count

Commit:
Combat blessing

Commit:
error fix

Commit:
Icon for precepts

Commit:
Palestone converting

Commit:
another icon color fix

Commit:
typo

Commit:
Various precepts fixes

Commit:
Hediff changed

Commit:
typo

Commit:
Sacrifice rebalance

Commit:
Product output chance for corpse sacrifice

MaxDorob committed:
1.6 merge

Commit:
Reworked reward system

Commit:
Changed reward curve

Commit:
Random factor

Commit:
Removed inspiration

Commit:
desc fix

Commit:
Ritual Precepts description

Commit:
Cleanup

Commit:
cleanup

Commit:
Vanilla thoughts description

Commit:
Small desc change

Commit:
text changes

Commit:
Old invoke blessing disabled by default

Commit:
Mod settings bug fix

Commit:
Fix

Commit:
typo

Commit:
Removed vanilla side-effect for positive rituals

Commit:
Different naming for non-lethal rituals

Commit:
Build fix

Commit:
Load folder fix

Commit:
Namespace fix

Commit:
Ref version

Commit:
Flexible effecters

Commit:
Increased defualt speed

Commit:
Comments

Commit:
Ability icons (thanks Mortis: https://steamcommunity.com/profiles/76561198836279025/)

Commit:
Mystical stone texture

Commit:
Ritual icons

MaxDorob committed:
Update 1.6/Defs/Building/Buildings_Art.xml

typo

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/HediffDefs/RitualHediffs.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update Languages/English/Keyed/Comps.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/ThoughtDefs/Thoughts_Ritual_Quality.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/ThoughtDefs/Thoughts_Ritual_Quality.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/ThoughtDefs/Thoughts_Ritual_Quality.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/ThoughtDefs/Thoughts_Ritual_Quality.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Text suggestions from Dr. Nolegs

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Update 1.6/Defs/MemeDefs/Skarnite.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

Commit:
Corrected report string

MaxDorob committed:
Psycast background (thanks @drakynwyrm https://cara.app/drakynwyrm)

MaxDorob committed:
Rename

MaxDorob committed:
Abilities and path texts

Commit:
Conversion names

Commit:
Psycast ritual naming

Commit:
Changed sacrifice ritual names

Commit:
Soulreap ritual naming

Commit:
Conversation -> Conversion (XML)

Commit:
Conversation -> Conversion (.cs)

Commit:
typo

Commit:
icon rename

Commit:
reverted dlls

Commit:
Revert "Assemblies"

This reverts commit 681b41f4edc05d7f14ca426fd022dba8b5e84542.

MaxDorob committed:
Hediff texts

MaxDorob committed:
Removed obsolete

MaxDorob committed:
Small bloody explosion

MaxDorob committed:
Big explosion changes

MaxDorob committed:
Huge explosion effecter

MaxDorob committed:
some optimization?

MaxDorob committed:
Cleanup (removed custom TargetFinder)

Commit:
typo

MaxDorob committed:
Merge branch 'RitualsAndPsycasts' into RitualsRework

MaxDorob committed:
Merge pull request #3 from MaxDorob/RitualsRework

Rituals Update

MaxDorob committed:
Merge branch 'RitualsAndPsycasts' into PsycastExpanded_Expansion

MaxDorob committed:
Merge pull request #4 from MaxDorob/PsycastExpanded_Expansion

Psycast Expanded expansion

Commit:
Rollbacked 1.5 dll

Commit:
Build fixes

MaxDorob committed:
End and fail triggers for soulreap blessing

Commit:
Changed bloodstones consumption

Commit:
Fixed psycast ritual

Commit:
Unlock bloodpath on ritual

MaxDorob committed:
Sacrificer required

MaxDorob committed:
Keyd strings

MaxDorob committed:
Xenotype conversion fixes

MaxDorob committed:
Bugfix

MaxDorob committed:
Revia gene request can be disabled via mod settings

MaxDorob committed:
Conversion starting quality

MaxDorob committed:
Removed obsolete class (not sure of the idea)

MaxDorob committed:
Disable old sacrifice/conversion

MaxDorob committed:
Removed offering from sacrifice. Cap first letter

MaxDorob committed:
cap

MaxDorob committed:
Buff for sacrifice

MaxDorob committed:
Some sacrifice rebalance

MaxDorob committed:
Release candidate DLLs

MaxDorob committed:
Various reaving reap improvements

DrMismagius commented on 1.6/Defs/Building/Buildings_Art.xml:
```diff
@@ -0,0 +1,78 @@
+  <ThingDef ParentName="ArtBuildingBase">
+    <defName>ReviaSkarnesSculpture</defName>
+    <label>skarne's sculpture</label>
+    <description>A sculpture of the nine-tailed bloody Goddess.</description>
```
```suggestion
    <description>A sculpture of the nine-tailed Bloody Goddess.</description>
```
(Missed this last time, 'Bloody Goddess' is a title, at least how we're using it.

DrMismagius commented on 1.6/Defs/MiscItemDefs/MysticalStone.xml:
```diff
@@ -0,0 +1,31 @@
+  <ThingDef ParentName="ResourceBase" MayRequire="Ludeon.Rimworld.Royalty">
+    <defName>Revia_MysticalStone</defName>
+    <label>mystical stone</label>
+    <description>A mystical stone sent by Skarne that contains a mystical power.</description>
```
```suggestion
    <description>A mystical bloodstone sent by Skarne that contains a mystical, psychic power.</description>
```
Feel free to ignore this one, this is just me writing for dummies and exact messaging.

DrMismagius commented on Ideology/Defs/PreceptDefs/Precepts_PrisonerExecution.xml:
```diff
@@ -0,0 +1,255 @@
+        <nullifyingTraits>
+          <li>Bloodlust</li>
+          <li>Psychopath</li>
+					<li>Cannibal</li>
```
GASP THERE IS AN EXTRA SPACE/TAB HERE D:<

DrMismagius commented on Languages/English/Keyed/Rituals.xml:
```diff
@@ -0,0 +1,10 @@
+	<RitualPredictedOutcomeDescReviaCount>Revia count</RitualPredictedOutcomeDescReviaCount>
+	<MessageRitualRolePawnIsAlreadyRevia>Already revia</MessageRitualRolePawnIsAlreadyRevia>
+	<MessageRitualRoleAnyRevianGeneRequired>Revian gene required</MessageRitualRoleAnyRevianGeneRequired>
+	<RitualTargetMysticalStoneInfo>Mystical stone required</RitualTargetMysticalStoneInfo>
```
```suggestion
	<RitualTargetMysticalStoneInfo>Mystical bloodstone required</RitualTargetMysticalStoneInfo>
```

DrMismagius commented on 1.6/Defs/Rituals/Ritual_Behaviors.xml:
```diff
@@ -0,0 +1,331 @@
+					<li Class="ReviaRace.Rituals.StageFailTrigger_NotEnoughCarriedThings">
+						<def>Revia_MysticalStone</def>
+						<count>1</count>
+						<desc>missing mystical stone</desc>
```
```suggestion
						<desc>missing mystical bloodstone</desc>
```

DrMismagius commented on 1.6/Defs/Rituals/Ritual_Behaviors.xml:
```diff
@@ -0,0 +1,331 @@
+					<li Class="ReviaRace.Rituals.StageFailTrigger_NotEnoughThings">
+						<def>Revia_MysticalStone</def>
+						<count>1</count>
+						<desc>missing mystical stone</desc>
```
```suggestion
						<desc>missing mystical bloodstone</desc>
```

DrMismagius commented on 1.6/Defs/PreceptDefs/Precepts_Ritual.xml:
```diff
@@ -0,0 +1,52 @@
+		<defName>Revia_PsycastRitual</defName>
+		<label>revia psycast blessing</label>
+		<tipLabelOverride>Revia psycast blessing</tipLabelOverride>
+		<description>A sacrificer will offer a mystical stone to Skarne for a connection to psychic powers.</description>
```
```suggestion
		<description>A sacrificer will offer a mystical bloodstone to Skarne for a link to psychic powers.</description>
```

DrMismagius commented on 1.6/Defs/MiscItemDefs/MysticalStone.xml:
```diff
@@ -0,0 +1,31 @@
+<Defs>
+  <ThingDef ParentName="ResourceBase" MayRequire="Ludeon.Rimworld.Royalty">
+    <defName>Revia_MysticalStone</defName>
+    <label>mystical stone</label>
```
```suggestion
    <label>mystical bloodstone</label>
```

Not even changing the def ID, just adding bloodstone to all of the relevant loc.

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_Bloodseek</defName>
+		<label>bloodseek</label>
+		<description>Teleports caster to the bleeding target.</description>
```
```suggestion
		<description>Attuning to nearby flowing blood and honing into to the location where it flows from, the caster may teleport to any currently bleeding target in the map.</description>
```

This is the first one for what I am actually here for, and where I will explain ideation.
Psycast descriptions, both in VPE and in Vanilla, tend to explain what they do using the structure of
Actions being taken
Mystical bullcrap
What Happens

Some are simpler, some don't follow this, and a lot are simply flowery, but I have put a decent amount of work into following or making use of that structure.

Again, before reaching live I need to make sure these look good in game, since some will have to be restructured.
I will include 2 requested psycast changes.

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_SkarneAvatar</defName>
+		<label>skarne's scion</label>
+		<description>Caster invokes to Skarne's real power. Caster becomes hungry for blood avatar of Skarne for short period.</description>
```
```suggestion
		<description>Starting a chant and building up the fury, this advanced form of the Bloodbath Circuit supercharges a Revia's already powerful connection to the Bloody Goddess through all nine of her tails, causing possession for a short time, becoming a berserk and extremely blood hungry avatar of Skarne. Slaying all who would dare oppose her viscous will, a true born scion of Skarne will slaughter all they can while under the influence, old allies being the last to be targeted for death. This extremely powerful and dangerous state of connection lasts psychic heat collapses this connection, the body becoming to damaged to fight, or simple time degrading the connection.</description>
```

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_Bloodbath</defName>
+		<label>bloodbath</label>
+		<description>Skarne guides caster through the battle.</description>
```
```suggestion
		<description>The most direct application of psychic power for any Revia is empowering her tails, her connection to Skarne. It being an obvious decision does not make it any less powerful.
By channeling a surge of psychic energy in a circuit through each of her tails, a Revia may leverage her that circuit at a moments notice to bathe in the blood of her adversaries. While active this technique immensely increases speed and allows for significantly more rapid attacks, with enough tails even allowing and then further empowering, the benefits of sanctified weaponry on any attack! The ensuing bloodbath becomes significantly bloodier the greater the level of Soul Reap.</description>
```

(Not sure if that's how you denote a new paragraph line in this)

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+		<castSound>VPE_BladeFocus_Cast</castSound>
+		<!--TODO: Change to some other sound-->
+		<targetMode>Self</targetMode>
+		<castTime>15</castTime>
```
```suggestion
		<castTime>5</castTime>
```

'A Moment's notice'

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_SkarnesProtection</defName>
+		<label>skarne's protection</label>
+		<description>Skarne may be merciful, closing bleeding wounds, often leaving a scar, like a mark...</description>
```
```suggestion
		<description>Skarne enjoys blood flowing freely, but may be merciful to her supplicants and their allies. Manipulating fresh blood on open wounds allows for their immediate healing, often leaving scars like a mark of her influence.</description>
```

(If you made this scale or it does scale with Psychic Sensitivity for heal% or even just heall then this is correct, if it doesn't heal (which would make it almost a worse sanguophage spell), heal becomes close, but it should probably be changed somewhat since scarring is a downside, I think we yapped about this a long while ago, either way, changed loc)

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_SkarnesCurse</defName>
+		<label>skarne's curse</label>
+		<description>Skarne curses the victim to bleed.</description>
```
```suggestion
		<description>By channeling the bloodthirsty viciousness that is integral to the Revia, the caster wills a curse on the target, not only vastly increasing vulnerability to bleeding and hemorrhaging , but causing open wounds to bursting out the victim over a short period dependent on how powerful the psychic connection.</description>
```

Does this scale on psychic sens on caster? Does this Scale on Tails?

'how powerful the psychic connection, and level of Soul Reap.' could be interchanged depending on that instead of current end. It probably should scale on psychic connection in any case.

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_PsycastBase">
+		<defName>Revia_BloodyExplosion</defName>
+		<label>bloody explosion</label>
+		<description>The victim's bleeding wounds explode.</description>
```
```suggestion
		<description>Manipulate energy in blood flowing from open wounds, snap your finger, and watch an explosion of gore fly out in all directions! The efficacy of which depends on how much the victim is bleeding at that point.</description>
```
"Snaps finger, blows up"
(I vote to reduce the cast time by a bunch, for this one, Dalia Mendoza moment)
https://i.namu.wiki/i/XRJeZ-LHjVmvMyKl7qrkO-JIuqjO9Bpu_YHTCMgdggBmgMy0tyTfRZ4jEhxccyy6RzDhC-NeM-skM_V1lLBn9a3aD3i1cvcEz86yyqIStFgyaK3QFV07skOXTQi9EAMAM-bLoAS49Y6ShCBObKv_w0g0DWYlZgT9MJJ2hldUpMY.webp

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+		<label>bloody explosion</label>
+		<description>The victim's bleeding wounds explode.</description>
+		<iconPath>UI/Bloodpath/BloodyExplosion</iconPath>
+		<castTime>40</castTime>
```
```suggestion
		<castTime>5</castTime>
```

MaxDorob replied:
Too fast, you can't snap your finger that fast

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+			<li Class="Revia_VanillaPsycastExpanded.AbilityExtension_BloodyExplosion">
+				<damageDef>Revia_BloodExplosionScratch</damageDef>
+				<radius>3.9</radius>
+				<damageAmount>8~12</damageAmount>
```
Radius and Damage of this specifically, could scale on psychic str, unless they do and I can't tell, although it doesn't need particularly extreme scaling.

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_WordOfAbilityBase">
+		<defName>Revia_ReavingReap</defName>
+		<label>reaving reap</label>
+		<description>Collect Skarne's rewards from corpses around.</description>
```
```suggestion
		<description>Skarne recognizes effort, and by expending psychic energy you will collect all nearby bloodstones or palestones from corpses as if you sacrificed them through labor. Simple, and time efficient.</description>
```

DrMismagius commented on PsycastExpanded/Defs/AbilityDefs/Bloodpath.xml:
```diff
@@ -0,0 +1,309 @@
+	<VEF.Abilities.AbilityDef ParentName="VPE_WordOfAbilityBase">
+		<defName>Revia_VermilionHarvest</defName>
+		<label>vermilion harvest</label>
+		<description>Spill blood of victim for Skarne's reward.</description>
```
```suggestion
		<description>There is power in blood, and the Bloody Goddess of all entities certainly knows how to use it. Converting blood directly into psyfocus is a harrowing and possibly fatal process... for whoever is being harvested. The larger the creature, the more blood that is taken, and killing the target all increase the reward. Friend, foe or even the caster herself, Skarne cares not how blood flows, only that it does.</description>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+  <HediffDef  ParentName="PsycastHediffBase">
+    <defName>Revia_Bloodbath</defName>
+    <label>bloodbath</label>
+    <description>Skarne guides me through the battle...</description>
```
```suggestion
    <description>I will bathe my {Number of Tails/SoulReap Level} tails in blood!</description>
```
(This skill Requires 3 tails, always plural!).
I don't know how to code that, or even if it's possible. if it isn't possible use
<description>I will bathe my empowered tails in blood!</description>

since 'The Bloodbath Circuit' is just throwing energy into those.

This is one of two skills I would like a significant advancement to, not including some silly casting speed changes, or questions on scaling of BOOM.

I am proposing a change to how this ability is balanced
Since it has no bonus from psychic strength, as it is right now, it has two concerns
Movement and Melee Dodge scale with eachother (Which is based, but double dippy)
The movement bonus, compared to other similar skills is limited (The Other Rush Skill is TRIPLE movement).

I propose a change, to amplify dexterity as key and make tail scaling really direct.
Moving to 0.1 per level of Tail (Level 3 is effective Minimum since you can't get this skill before then) so 0.3 Start
Armor Penetration from Melee (Ignoring Armor through Sheer Speed, How Anime!)
Weapon/Cooldowns (As is)
Sanctified Bonus (Level 6 upwards, as is)

MaxDorob replied:
```suggestion
    <description>I will bathe my empowered tails in blood!</description>
```
I'm lazy to implement it, lol

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.05</postFactor>
```
```suggestion
						<postFactor>1.1</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.02</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.1</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+				<capMods>
+					<li>
+						<capacity>Moving</capacity>
+						<postFactor>1.10</postFactor>
```
```suggestion
						<postFactor>1.20</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+					</li>
+				</capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.06</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.2</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.15</postFactor>
```
```suggestion
						<postFactor>1.3</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.11</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.3</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.18</postFactor>
```
```suggestion
						<postFactor>1.4</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.16</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.4</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.22</postFactor>
```
```suggestion
						<postFactor>1.5</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.20</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.5</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.28</postFactor>
```
```suggestion
						<postFactor>1.6</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.25</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.6</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+				<capMods>
+					<li>
+						<capacity>Moving</capacity>
+						<postFactor>1.35</postFactor>
```
```suggestion
						<postFactor>1.7</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+					</li>
+				</capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.30</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.7</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+					</li>
+				</capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.35</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.8</MeleeArmorPenetration>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+				<capMods>
+					<li>
+						<capacity>Moving</capacity>
+						<postFactor>1.40</postFactor>
```
```suggestion
						<postFactor>1.80</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+        <capMods>
+          <li>
+            <capacity>Moving</capacity>
+						<postFactor>1.45</postFactor>
```
```suggestion
						<postFactor>1.9</postFactor>
```

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/Bloodbath.xml:
```diff
@@ -0,0 +1,158 @@
+          </li>
+        </capMods>
+				<statFactors>
+					<MeleeDodgeChance>1.4</MeleeDodgeChance>
```
```suggestion
					<MeleeArmorPenetration>1.9</MeleeArmorPenetration>
```

(End of Stat Changes)
A Nine Tailed Revia casting will end at a 1.9 Mult Speed and Armor Pen (VS 2 with Skarne and 2.5 with Skarne)
You could possibly make Skarne's Movement 2.5 as well if you want parity, I think this change is valid.

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/SkarneAvatar.xml:
```diff
@@ -0,0 +1,38 @@
+    <defName>Revia_SkarneAvatar</defName>
+    <label>skarne avatar</label>
+		<hediffClass>HediffWithComps</hediffClass>
+    <description>Skarne owns this body.</description>
```
https://www.youtube.com/watch?v=7ki7CLMTlYY

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/SkarneAvatar.xml:
```diff
@@ -0,0 +1,38 @@
+					<MeleeCooldownFactor>0.3</MeleeCooldownFactor>
+					<MeleeArmorPenetration>2.5</MeleeArmorPenetration>
+					<MeleeHitChance>1.6</MeleeHitChance>
+					<MeleeDodgeChance>2.2</MeleeDodgeChance>
```
```suggestion
					<Revia_LifeLeechImpact>0.9</Revia_LifeLeechImpact>
					<StaggerDurationFactor>0</StaggerDurationFactor>
```

Skarne simply wants blood to flow!

(Similar reasoning to my greater Bloodbath change,  Movement applies to these already, and given all the DR bonuses, the Sanctified Liferip and stagger immunity may be the certified gamer move.)

DrMismagius commented on PsycastExpanded/Defs/MentalStateDefs/MentalStates_SkarneAvatar.xml:
```diff
@@ -0,0 +1,34 @@
+    <moodRecoveryThought>Catharsis</moodRecoveryThought>
+    <blockNormalThoughts>true</blockNormalThoughts>
+    <nameColor>(0.9,0.2,0.5)</nameColor>
+    <beginLetter>{0} has gone berserk.\n\n[PAWN_pronoun] will attack anyone [PAWN_pronoun] sees.</beginLetter>
```
```suggestion
    <beginLetter>{0} has gone berserk from possession.\n\n[PAWN_pronoun] will focus on enemies yet attack anyone [PAWN_pronoun] sees.</beginLetter>
```
(Need to see in game word wrapping, slightly more descriptive)

DrMismagius commented on PsycastExpanded/Defs/PsycasterPathDefs/Paths.xml:
```diff
@@ -0,0 +1,14 @@
+  <VanillaPsycastsExpanded.PsycasterPathDef>
+    <defName>Revia_BloodPath</defName>
+    <label>bloodpath</label>
+    <description>Revia psycasts are focused on fighing, especially against flesh-based enemies.</description> 
```
```suggestion
    <description>Revia psycasts are focused on exalting the Bloody Goddess Skarne, fighting and wounding, especially against that which can bleed. 
 Those on the bloodpath are uniquely attached to Skarne through the Soul Reap, much like how normal psycasters are attached to their archotech networks. Bloodpath psycasters are usually psychically linked Revia seeking nothing less than the total reverence of the Bloody Goddess and their own strength. The powers granted by the Bloodpath tend to scale not only with Psychic sensitivity, but also with the amount of tails the Revia have acquired through the Soul Reap. Blood! Blood! Blood!</description> 
```

DrMismagius commented on PsycastExpanded/Defs/PsycasterPathDefs/Paths.xml:
```diff
@@ -0,0 +1,14 @@
+    <background>UI/Bloodpath/Background</background>
+    <altBackground>UI/Bloodpath/Background</altBackground>
+    <tab>Psycasts</tab>
+    <tooltip>Revia psycasts are focused on fighing, especially against flesh-based enemies. Skarne gives power Her scions.</tooltip>
```
```suggestion
    <tooltip>Revia psycasts are focused on exalting the Bloody Goddess Skarne, fighting and wounding, especially against that which can bleed.</tooltip>
```

Not sure exactly which is supposed to be the long description, guessing this one, swap where needed.

DrMismagius commented on PsycastExpanded/Defs/Stats/Revia_VPE_Stats.xml:
```diff
@@ -0,0 +1,16 @@
+  <StatDef>
+    <defName>Revia_PsyfocusPerKill</defName>
+    <label>psyfocus gain</label>
+    <description>Amount of psyfocus gain per kill.</description>
```
```suggestion
    <description>Psyfocus on kill.</description>
```
(More succinct, I also could swear this stat already exists somewhere)

DrMismagius reviewed (changes requested):
A Few Changes to older stuff, A Few small Stat Changes, One Request for Info to possible rewrite, and a large change to on skill.

Hope this helps, was helpful and continues to be helpful.

These do need to be a bit tested in game.

Thank you for being you.

MaxDorob committed:
Apply suggestions from code review

Various text changes from Dr.Nolegs

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Bloodbath rebalance

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Reduced time of bloodbath to 30 real seconds

MaxDorob committed:
Various text changes

MaxDorob committed:
Reduced chance of scar, maxed count of scars, ability now heals injury a bit

MaxDorob committed:
Release candidate DLL 2

MaxDorob committed:
Update 1.6/Defs/Rituals/Ritual_Behaviors.xml

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

MaxDorob committed:
Stagger removed

Co-authored-by: DrMismagius <118777948+DrMismagius@users.noreply.github.com>

DrMismagius commented on PsycastExpanded/Defs/HediffDefs/SkarneAvatar.xml:
```diff
@@ -22,7 +22,8 @@
 					<MeleeArmorPenetration>2.5</MeleeArmorPenetration>
 					<MeleeHitChance>1.6</MeleeHitChance>
-					<MeleeDodgeChance>2.2</MeleeDodgeChance>
+					<Revia_LifeLeechImpact>0.9</Revia_LifeLeechImpact>
```
1.5 or 2

Yeah so uh, this was an error since I thought this would add 0.9, Skarne isn't nerfing her own kit!

DrMismagius committed:
Update Bloodpath.xml

Minor Rebalances, including increasing heat costs across the board for the most part.

Noted Bloodbath changes, will include a re-nerfing of speed and an increase in cost (To Reduce Doubledip Synergies) and heat while reintroducing a solid duration.

Increasing heat and reducing Psycost of Vermillion Harvest (which regens Psycost as a benefit, so reducing the required floor is wise)

DrMismagius committed:
Update Bloodbath.xml

Changes in relation to main ability changes.
Includes a mild nerf/reshuffle of level 9

DrMismagius committed:
Update SkarneAvatar.xml

Moderate Changes, likely an overall Nerf but a warranted one.

Reduces Pain Res Buff, increases Melee Damage and Hit Chance, Reduces Cooldown bonus, slightly reduces Melee Armor Pen and makes lifeleech a bonus (whoops), while reducing the damage resistance by a marked amount.

You may also notice that this makes the factors add up to 9.
(Runs away)

MaxDorob committed:
Merge pull request #6 from DrMismagius/patch-3

Update SkarneAvatar

MaxDorob committed:
Merge pull request #8 from DrMismagius/patch-1

Update Bloodpath.xml

MaxDorob committed:
Merge pull request #7 from DrMismagius/patch-2

Nerfed bloodbath

DrMismagius committed:
Update 2 Bloodpath.xml

Shortening where I can, certain descriptions were overlong.

The Cast times for Reap and Harvest were definitely on the exceptionally long time, and are still fairly lengthy, but now slightly less so.

DrMismagius committed:
Update Ritual_Outcomes.xml

DrMismagius committed:
Update ReviaRaceMod.cs

Please?

DrMismagius committed:
Update ReligiousCeremonies.xml

Gendered the Language

DrMismagius committed:
Update SanctifiedWeaponry.xml

Gendered 2

DrMismagius committed:
Update BloodBenediction.xml

Skarne this time

DrMismagius committed:
Update MysticalStone.xml

Slight changes, mostly due to the fact this can spawn in artifact pools, so allows non-Revia reading.

Commit:
Rituals desc (from Dr. Nolegs)

Commit:
Reveted fail chance

MaxDorob committed:
Cost growth reset

MaxDorob committed:
Error fix and sculpture naming rule pack

MaxDorob committed:
NRE fix

MaxDorob committed:
Changed psycast level up system

MaxDorob committed:
DLL release candidate 3

MaxDorob committed:
Reduced defualt growth factor, typo

MaxDorob committed:
dll RC

MaxDorob committed:
Merge pull request #9 from DrMismagius/patch-1

Various changes

MaxDorob committed:
Removed comments

MaxDorob committed:
Merge branch 'RitualsAndPsycasts' into RitualsAndPsycasts

MaxDorob committed:
Build error fix

MaxDorob committed:
Merge pull request #10 from DrMismagius/RitualsAndPsycasts

Various changes

MaxDorob committed:
Removed obsolete setting

MaxDorob committed:
Another RC dll

MaxDorob committed:
We don't want a chicken icon for Skarne

MaxDorob committed:
Name maker fix

MaxDorob committed:
Revia psycasters in raid, hell yeah!

MaxDorob committed:
Let AI cast bloodbath and avatar

MaxDorob committed:
Kinda warning fix

Commit:
Post Skarne's Avatar debuff

Commit:
settings page fix

Commit:
Disabled vfx from ideology

Commit:
Another RC dll

DrMismagius committed:
Update Comps.xml

Suggested Changes to Sacrifice wording
Capitalization Fixes for proper nouns, and using Skarne's Feminine title.

DrMismagius committed:
Update AbilityExtension_VermilionHarvest.cs

Part of small changes to Verm harvest - now returns 4x psyfocus while still pounding you with Psy Heat.

Increasing the cast time somewhat, and including the rewording.

DrMismagius committed:
Update Bloodpath.xml

Words Words Words.
Many small changes
Lots more wording changes, lots more wording changes, waay to many wording changes.
Words Words Words.

DrMismagius committed:
Update SkarneAvatar.xml

Clinical Tone for the Hediff.

MaxDorob committed:
Force to spawn with level by extension

MaxDorob committed:
Soul Tax desc

MaxDorob committed:
MORE TEXT CHANGES (and some balance changes)

MaxDorob committed:
Increased psyfocus gain per death from vermilion harvest

MaxDorob committed:
More text changes

MaxDorob committed:
Blessed weapons with psyfocus gain

MaxDorob committed:
dll

MaxDorob committed:
fixed damn bug

MaxDorob committed:
VPE addon dll
